crates/ruma-events/src/room/encrypted/unstable_state.rs

```diff
@@ -0,0 +1,42 @@
+//! Types for `m.room.encrypted` state events, as defined in [MSC3414][msc].
+//!
+//! [msc]: https://github.com/matrix-org/matrix-spec-proposals/blob/travis/msc/encrypted-state/proposals/3414-encrypted-state.md
+use ruma_macros::EventContent;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    room::encrypted::EncryptedEventScheme, False, PossiblyRedactedStateEventContent,
+    StateEventType, StaticEventContent,
+};
+
+/// The content of an `m.room.encrypted` state event.
+#[derive(Clone, Debug, Deserialize, Serialize, EventContent)]
+#[cfg_attr(not(ruma_unstable_exhaustive_types), non_exhaustive)]
+#[ruma_event(type = "m.room.encrypted", kind = State, state_key_type = String, custom_possibly_redacted)]
+pub struct StateRoomEncryptedEventContent {
+    /// Algorithm-specific fields.
+    #[serde(flatten)]
+    pub scheme: EncryptedEventScheme,
+}
+
+/// The PossiblyRedacted form of [StateRoomEncryptedEventContent].
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[cfg_attr(not(ruma_unstable_exhaustive_types), non_exhaustive)]
+pub struct PossiblyRedactedStateRoomEncryptedEventContent {
+    /// Algorithm-specific fields.
+    #[serde(flatten, skip_serializing_if = "Option::is_none")]
+    pub scheme: Option<EncryptedEventScheme>,
+}
+
+impl StaticEventContent for PossiblyRedactedStateRoomEncryptedEventContent {
+    const TYPE: &'static str = "m.room.encrypted";
+    type IsPrefix = False;
```

Comment:
nit: To reduce duplication and avoid possible copy-paste errors, let's use the values from `StateRoomEncryptedEventContent`:

```suggestion
    const TYPE: &'static str = StateRoomEncryptedEventContent::TYPE;
    type IsPrefix = <StateRoomEncryptedEventContent as StaticEventContent>::IsPrefix;
```